core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -178,6 +179,10 @@ public static State fromString(String state) {
     public static final Setting<Boolean> INDEX_SHADOW_REPLICAS_SETTING =
         Setting.boolSetting(SETTING_SHADOW_REPLICAS, false, Property.IndexScope);
 
+    public static final String SETTING_PARTITION_SIZE = "index.partition_size";
```

Comment:
I don't have a better suggestion at the moment, but I wish this name included something like `shard` or `routing` to make it clearer what it is about.